### PR TITLE
React & Redux group project - Space Travelers' Hub

### PR DESCRIPTION
**Pull request by CindyDorantes**
#### Config & basic setup :clipboard:
- Create React App.
- Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://reactrouter.com/web/guides/quick-start).
- Download the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- Create routes and view components (Rockets, Missions, My Profile, Dragons). Use `<NavLink />` for the page navigation links and style active class to indicate which section/page user is currently on (underline active navigation link).
- Create directories for all Redux state slice files (rockets, missions, dragons).

#### Redux: Fetch data and update Redux store :telephone_receiver:
- Upon first render fetch data from the SpaceX API endpoints.
- Once the data are fetched, dispatch an action to store the selected data in Redux store.
- Make sure you only dispatch those actions once and do not add data to store on every re-render

#### Render UI:lists :art:
- Use `useSelector()` Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes.
- You can style the whole application "by hand" or you could use [React Bootstrap](https://react-bootstrap.github.io/).
- Render a list of rockets (as per design). For the image of a rocket use the first image in the array of `flickr_images`.
- Render a table with the missions' data (as per design).
- Render a list of dragons (as per design). For the image of a dragon use the first image in the array of `flickr_images`.

#### Redux: Write actions and reducers for booking rockets/dragons and joining missions :rocket:
- When a user clicks the "Reserve rocket" button or "Reserve dragon" button, action needs to be dispatched to update the store. You need to get the ID of the reserved rocket and update the state. Remember you mustn't mutate the state. Instead, you need to return a new state object with all rockets, but the selected rocket will have an extra key `reserved` with its value set to `true`. 
- Make sure you place all your logic in the reducer. In the React view file, you should only dispatch the action with the correct rocket ID as an argument.
- Create a reducer and action dispatcher for the "Join Mission" button. The logic here is practically the same as with rockets - you need to pass the mission's ID to the corresponding action and update the missions' state with the selected mission having a new key/value - `reserved: true`.

#### Redux: Write actions and reducers for canceling rockets/dragons and leaving missions :no_entry:
- Here you need to follow the same logic as with the "Reserve rocket"/"Reserve dragon" and "Join mission" - but you need to set the `reserved` key to `false`. 
- Dispatch these actions upon click on the corresponding buttons.

#### Render UI: conditional components rendering :interrobang:
- Rockets that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve rocket" (as per design) .
- Dragons that have already been reserved should show a "Reserved" badge and "Cancel reservation" button instead of the default "Reserve dragon" (as per design).
- Missions that the user has joined already should show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button (as per design).
- Rockets/Dragons and Missions should use the React conditional rendering syntax.

#### Render UI: My Profile section :checkered_flag:
- Compose two/three column layout and list ONLY the rockets/dragons reserved and missions joined by the user (as per design):
    - Render a list of all joined missions (use `filter()`).
    - Render a list of all reserved rockets (use `filter()`).
    - Render a list of all reserved dragons (use `filter()`).

#### Technical set up** :wrench:
- Set up the repository on GitHub and use Gitflow.
- Set up React and Redux.
- Set up testing libraries(React Testing Library and Jest).
- Do not preserve user data

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+import thunk from 'redux-thunk';
+import missionReducer from './Mission';
```
Comment (tufoinnkuo10) [inline src/redux/configureStore.js:3]: - [ ] please add the mission file in its own folder for good code practices

File: src/App.js [hunk 2]
```diff
@@ -1,23 +1,24 @@
...
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+    <div>
+      <Navbar />
```
Comment (tufoinnkuo10) [inline src/App.js:13]: - [ ] please align the images for the rockets and dragons. so they all can be in order
 - [ optional] you can also make the reserved text to show red so a user takes it serious and easily notice that is has been reserved 
Screenshot (413)_LI
Screenshot (414)_LI

good example
Screenshot (415)

File: README.md [hunk 3]
```diff
@@ -1,70 +1,71 @@
...
-Launches the test runner in the interactive watch mode.\
-See the section about [running tests](https://facebook.github.io/create-react-app/docs/running-tests) for more information.
+## GETTING STARTED
 
-### `npm run build`
+Clone this repo : git@github.com:CindyDorantes/Space-Travelers-Hub.git
 
-Builds the app for production to the `build` folder.\
-It correctly bundles React in production mode and optimizes the build for the best performance.
+After cloning this project, run the following commands:
```
Comment (tufoinnkuo10) [inline README.md:23]: - [ ] kindly add a description as well on how to run your test

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
Hi @Team

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Git flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ app well-deployed
✅ app works as expected

Required Changes ♻️
Check the comments under the review

tufoin

Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: src/App.js (hunk 2 above)
Comment (rloterh) [inline src/App.js:13]: - [ ] Kindly ensure that the images are of equal dimensions and are aligned, as informed in the previous code review.

**Your design:**
image


**The original design:**
image

File: README.md (hunk 3 above)
Comment (rloterh) [inline README.md:23]: - [ ] Kindly include instructions in your ReadMe on how to run your test, as informed in the previous review.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi Team, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 🚦 🚦 🚦  

## Highlights 👍  
- Implemented required components
- State management with redux is properly done
- Data is properly fetched from endpoint via store
- Navigations works as intended
- Good unit test coverage with React testing library



## Required Changes ♻️
_Kindly check the comments under the review._


<br>
Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 3 above)
Comment (CindyDorantes) [reply README.md:25]: I'm really sorry and so embarrassed. Those changes were implemented and committed but I didn't realize I didn't push the code.

**Review by smunozmo — APPROVED** [review]
Hi @CindyDorantes, 👋 

## Highlights 👏 

- Excellent job, another cool project for your portfolio, congrats on it! 🎉 
- All the suggested changes were made, perfect! 😎 
- Great work using the right GitHub Flow.
- Your README looks professional, well done!
- Commits are descriptive, good work.
- Linters are working properly, great job!

Your project is complete!  There is nothing else to say other than... it's time to merge it! 🚀 
Very nice work, congratulations! 🎉 

# APPROVED! 💯  💯  💯 




Cheers and Happy coding! 🤓 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

✅  Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**